### ISSUE TITLE
A recorded but non certified Cardano transaction creates an error in prover

### ISSUE THREAD

**Issue by jpraynaud** [input-output-hk/mithril#1819]
## Why
A Cardano transaction recorded but non certified yet make the prover return a global error when it should just mark the transaction as non certified:
```
Internal error of the Aggregator

Caused by:
    MKMap could not replace non-existing key
```

The error occurs with a transaction:
- recorded
- not certified yet
- or in the last block of the last signed block range

## What
Make the prover succeed even if a transaction is recorded and not certified yet (the same way as for non recorded yet transaction).

## How

- [x] Fix the error with #1785 
- [x] ~~Benchmark the performances and verify there is no regression~~

**Comment by jpraynaud** [comment]
Blocked by #1785 which should fix the problem

**Comment by jpraynaud** [comment]
Fixed with #1785
